platform/darwin/src/http_file_source.mm

```diff
@@ -196,24 +196,45 @@ void cancel() {
 
 HTTPFileSource::~HTTPFileSource() = default;
 
+MGL_EXPORT
+NSURL *resourceURLWithAccountType(const Resource& resource, NSInteger accountType) {
+    
+    NSURL *url = [NSURL URLWithString:@(resource.url.c_str())];
+    
+#if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
+    if (accountType == 0 &&
+        ([url.host isEqualToString:@"mapbox.com"] || [url.host hasSuffix:@".mapbox.com"])) {
+        NSURLComponents *components = [NSURLComponents componentsWithURL:url resolvingAgainstBaseURL:NO];
+        NSURLQueryItem *accountsQueryItem = [NSURLQueryItem queryItemWithName:@"sku" value:MGLAccountManager.skuToken];
+        
+        NSMutableArray *queryItems = [NSMutableArray arrayWithObject:accountsQueryItem];
+        
+        // offline here
+        if (resource.usage == Resource::Usage::Offline) {
+            [queryItems addObject:[NSURLQueryItem queryItemWithName:@"offline" value:@"true"]];
+        }
+        
+        if (components.queryItems) {
+            [queryItems addObjectsFromArray:components.queryItems];
+        }
+        
+        components.queryItems = queryItems;
+        url = components.URL;
+    }
+#else
+    (void)accountType;
```

Comment:
(This is a stub pending a resolution to https://github.com/mapbox/mapbox-gl-native/issues/14835#issuecomment-499297557.)